src/sinks/http.rs

```diff
@@ -232,9 +231,16 @@ impl HttpSink for HttpSinkConfig {
             Encoding::Text => "text/plain",
             Encoding::Ndjson => "application/x-ndjson",
             Encoding::Json => {
-                body.insert(0, b'[');
-                body.pop(); // remove trailing comma from last record
-                body.push(b']');
+                let message = body.split();
+                body.put_u8(b'[');
+                if !message.is_empty() {
+                    // TODO: Prepend before building a request to eliminate the
+                    // additional copy here.
```

Comment:
Do we want to do this before merging or save for later?

Reply:
I looked briefly into it. Couldn't find a solution after digging through several wrappers around `HttpSink`. Would be happy to defer since I don't think it will be only a couple of lines change.

Reply:
Worth noting that `Vec::insert` does something a but more clever here: https://github.com/rust-lang/rust/blob/775e480722c7aba6ff4ff3ccec8c1f4639ae7889/library/alloc/src/vec/mod.rs#L1354-L1367.

However, I decided not to go for an `unsafe` solution for the moment.

Reply:
Makes sense, it's something we can come back to if it shows up as being significant in profiling. If you've explored it already, maybe worth dropping some notes into an issue and linking from the comment?

Reply:
I don't think we have any soak tests that exercise this encoding? I'd suggest we add one to measure the impact of this before merging.

Reply:
Adressed in 201e2724f94bf82c3c4cfc01f2896cf5a303d8af.

Reply:
Kicked off 91df7e10e3a6ad13547ada0f71b0613fe100413b where I copied the implementation of `Vec::insert`. Let's see what the soaks say.